lib/api/controllers/memoryStorageController.js

```diff
@@ -666,40 +836,71 @@ function extractArgumentsFromRequestForZAdd (request) {
  * @returns {*[]}
  */
 function extractArgumentsFromRequestForZInterstore (request) {
-  var args = [
-      request.input.body.destination
-    ],
-    keys = [];
+  let args = [request.input.resource._id];
 
-  if (request.input.resource._id) {
-    keys.push(request.input.resource._id);
-  }
-  if (request.input.body.keys !== undefined) {
-    assertBodyAttributeType(request, 'keys', 'array');
+  kassert.assertHasId(request);
+  kassert.assertHasBody(request);
+  kassert.assertBodyAttributeType(request, 'keys', 'array');
 
-    keys = keys.concat(request.input.body.keys);
+  if (request.input.body.keys.length === 0) {
+    throw new BadRequestError(`ms:${request.input.action} At least 1 source key must be provided`);
   }
 
-  args.push(keys.length);
-  args = args.concat(keys);
+  args.push(request.input.body.keys.length);
+  args.push(...request.input.body.keys);
```

Comment:
This could lead to an unexpected error if the user input `keys` is big enough:
```
let
  a = [],
  b = [];

for (let i = 0; i < 1000000; i++) {
  a.push(i);
}

b.push(...a);

RangeError: Maximum call stack size exceeded
    at repl:1:3
    at sigintHandlersWrap (vm.js:22:35)
    at sigintHandlersWrap (vm.js:96:12)
    at ContextifyScript.Script.runInThisContext (vm.js:21:12)
    at REPLServer.defaultEval (repl.js:313:29)
    at bound (domain.js:280:14)
    at REPLServer.runBound [as eval] (domain.js:293:12)
    at REPLServer.<anonymous> (repl.js:503:10)
    at emitOne (events.js:101:20)
    at REPLServer.emit (events.js:188:7)
```